compiler/rustc_codegen_rmc/src/compiler_interface.rs

```diff
@@ -164,13 +161,15 @@ impl CodegenBackend for GotocCodegenBackend {
 
             let symtab_filename = base_filename.with_extension("symtab.json");
             debug!("output to {:?}", symtab_filename);
-            let mut out_file = ::std::fs::File::create(&symtab_filename).unwrap();
-            write!(out_file, "{}", result.symtab.to_irep().to_json().pretty().to_string()).unwrap();
+            let out_file = ::std::fs::File::create(&symtab_filename).unwrap();
+            let writer = BufWriter::new(out_file);
+            serde_json::to_writer(writer, &result.symtab.to_irep()).unwrap();
 
             let type_map_filename = base_filename.with_extension("type_map.json");
             debug!("type_map to {:?}", type_map_filename);
-            let mut out_file = ::std::fs::File::create(&type_map_filename).unwrap();
-            write!(out_file, "{}", result.type_map.to_json().pretty().to_string()).unwrap();
+            let out_file = ::std::fs::File::create(&type_map_filename).unwrap();
+            let writer = BufWriter::new(out_file);
+            serde_json::to_writer(writer, &result.type_map).unwrap();
```

Comment:
We do this bit more than once. Should it be a helper function?

Reply:
Would also help make sure that temporaries were dropped as soon as the helper function exited.

Reply:
ok